go-packages/meep-metric-store/metric-store.go

```diff
@@ -0,0 +1,243 @@
+		err = ms.connectDB(addr)
+		if err != nil {
+			log.Warn("Failed to connect to InfluxDB. Retrying... Error: ", err)
+		}
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	// Set store to use
+	err = ms.SetStore(name)
+	if err != nil {
+		log.Error("Failed to set store: ", err.Error())
+		return nil, err
+	}
+
+	log.Info("Successfully connected to Influx DB")
+	return ms, nil
+}
+
+func (ms *MetricStore) connectDB(addr string) error {
+	if addr == "" {
+		ms.addr = "http://meep-influxdb:8086"
+	} else {
+		ms.addr = addr
+	}
+	log.Debug("InfluxDB Connector connecting to ", ms.addr)
+
+	client, err := influxclient.NewHTTPClient(influxclient.HTTPConfig{Addr: ms.addr, InsecureSkipVerify: true})
+	if err != nil {
+		log.Error("InfluxDB Connector unable to connect ", ms.addr)
+		return err
+	}
+	defer client.Close()
+
+	_, version, err := client.Ping(1000 * time.Millisecond)
+	if err != nil {
+		log.Error("InfluxDB Connector unable to connect ", ms.addr)
+		return err
+	}
+
+	ms.client = &client
+	ms.connected = true
+	log.Info("InfluxDB Connector connected to ", ms.addr, " version: ", version)
+	return nil
+}
+
+// SetStore -
+func (ms *MetricStore) SetStore(name string) error {
+	// Set current store. Create new DB if necessary.
+	if name != "" {
+		q := influxclient.NewQuery("CREATE DATABASE "+name, "", "")
+		_, err := (*ms.client).Query(q)
+		if err != nil {
+			log.Error("Query failed with error: ", err.Error())
+			return err
+		}
+		ms.name = name
+	}
+	return nil
+}
+
+// Flush
+func (ms *MetricStore) Flush() {
+	// Make sure we have set a store
+	if ms.name == "" {
+		return
+	}
+
+	// Create Store DB if it does not exist
+	q := influxclient.NewQuery("DROP SERIES FROM /.*/", ms.name, "")
+	response, err := (*ms.client).Query(q)
+	if err != nil {
+		log.Error("Query failed with error: ", err.Error())
+	}
+	log.Info(response.Results)
+}
+
+// SetMetric - Generic metric setter
+func (ms *MetricStore) SetMetric(metric string, tags map[string]string, fields map[string]interface{}) error {
+	// Make sure we have set a store
+	if ms.name == "" {
+		err := errors.New("Store name not specified")
+		return err
+	}
+
+	// start = time.Now()
+
+	// Create a new point batch
+	bp, _ := influxclient.NewBatchPoints(influxclient.BatchPointsConfig{
+		Database:  ms.name,
+		Precision: "us",
+	})
+
+	// Create a point and add to batch
+	pt, err := influxclient.NewPoint(metric, tags, fields)
+	if err != nil {
+		log.Error("Failed to create point with error: ", err)
+		return err
+	}
+	bp.AddPoint(pt)
+
+	// Write the batch
+	err = (*ms.client).Write(bp)
+	if err != nil {
+		log.Error("Failed to write point with error: ", err)
+		return err
+	}
+
+	// logTimeLapse("SetMetric duration: ")
+
+	return nil
+}
+
+// GetMetric - Generic metric getter
+func (ms *MetricStore) GetMetric(metric string, tags map[string]string, fields []string, duration string, count int) (values []map[string]interface{}, err error) {
```

Comment:
Isn't this a limiting factor to assume a duration always based on Now()-XX.. preventing from doing searches as part of a group of historical data.. ?

Reply:
Yes, this was done to simplify the interface for now. The API should be implemented the same way as well for now, at least until we have a need to time ranges. Another reason for this simplification is to avoid timezone differences & out of sync times.